src/hotspot/share/opto/addnode.cpp

```diff
@@ -91,9 +91,17 @@ static bool commute(Node *add, bool con_left, bool con_right) {
 
   PhiNode *phi;
   // Check for tight loop increments: Loop-phi of Add of loop-phi
-  if( in1->is_Phi() && (phi = in1->as_Phi()) && !phi->is_copy() && phi->region()->is_Loop() && phi->in(2)==add)
+#ifdef ASSERT
+  if (in1->is_Phi())
+    assert(!in1->as_Phi()->is_copy(), "in1 cannot be a copy");
+#endif
+  if( in1->is_Phi() && (phi = in1->as_Phi()) && phi->region()->is_Loop() && phi->in(2)==add)
     return false;
-  if( in2->is_Phi() && (phi = in2->as_Phi()) && !phi->is_copy() && phi->region()->is_Loop() && phi->in(2)==add){
+#ifdef ASSERT
+  if (in2->is_Phi())
+    assert(!in2->as_Phi()->is_copy(), "in2 cannot be a copy");
+#endif
+  if( in2->is_Phi() && (phi = in2->as_Phi()) && phi->region()->is_Loop() && phi->in(2)==add){
```

Comment:
Since the row is touched - please add whitespace where its missing:
"phi->in(2)==add" -> "phi->in(2) == add"